fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java

```diff
@@ -106,35 +106,6 @@ public Subspace getSecondarySubspace() {
     public abstract <M extends Message> CompletableFuture<Void> update(@Nullable FDBIndexableRecord<M> oldRecord,
                                                                        @Nullable FDBIndexableRecord<M> newRecord);
 
-    /**
-     * Remove or add a uniqueness violation within the database. This is used to keep track of
-     * uniqueness violations that occur when an index is in write-only mode, both during
-     * the built itself and by other writes. This means that the writes will succeed, but
-     * it will cause a later attempt to make the index readable to fail.
-     * @param valueKey the indexed key that is (apparently) not unique
-     * @param primaryKey the primary key of one record that is causing a violation
-     * @param existingKey the primary key of another record that is causing a violation (or <code>null</code> if none specified)
-     * @param remove <code>true</code> if removing the violation and <code>false</code> if adding it. When
-     * <code>true</code>, it is favored use {@link #removeUniquenessViolationsAsync(Tuple, Tuple)} instead.
-     */
-    @Nonnull
-    public abstract void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple primaryKey, @Nullable Tuple existingKey, boolean remove);
```

Comment:
I suppose it's reasonable that `IndexMaintainer` wouldn't have a method for updating uniqueness violations and that that would be something that only lives within the implementations, though it wasn't entirely obvious to me that that's right.